src/riscv/linux/init.c

```diff
@@ -247,7 +247,7 @@ void cpuinfo_riscv_linux_init(void) {
 	const size_t max_processor_id = 1 +
```

Comment:
Just curious, what is the downside of casting this to `uint32_t`? and this way you don't need to cast that will throw compiler warning on 32-bit platform?

Also, I don't know much about RISC-V, is this a defined behavior the plafrom/compiler it overflows to 0?
If I'm not mistaken, C standard does not define how system should behave in this case

Reply:
I could've sworn there was a concrete reason I didn't do this, but it's been too long since I originally wrote this :/ - however, trying this out right now, I don't see any obvious harm and it seems to work. I'll put up that change shortly.

In regards to the overflow, I took a brief look into what the deal was here - it turns out, C **does** define the system behavior for integer overflow, but only when that integer is *unsigned**! The specific text is:

"A computation involving unsigned operands can never overﬂow, because a result that cannot be represented by the resulting unsigned integer type is reduced modulo to the number that is one greater than the largest value that can be represented by the resulting type."

There are some nice StackOverflow posts about this too, but the tl;dr if it's unsigned, it's guaranteed to wrap-around to 0 on overflow.